### PR TITLE
Bump opentelemetry-go to v1.28.0/v0.50.0

### PR DESCRIPTION
**Pull request by mboersma**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Updates opentelemetry-go to the [v1.28.0/v0.50.0](https://github.com/open-telemetry/opentelemetry-go/releases/tag/v1.28.0) release.

**Which issue(s) this PR fixes**:

Refs #4860

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by mboersma** [comment]
/hold

This needs manual testing with Tilt.

File: go.mod [hunk 1]
```diff
@@ -172,7 +172,7 @@ require (
 	github.com/peterbourgon/diskv v2.0.1+incompatible // indirect
 	github.com/pkg/browser v0.0.0-20240102092130-5ac0b6a4141c // indirect
 	github.com/prometheus/client_model v0.6.1 // indirect
-	github.com/prometheus/common v0.54.0 // indirect
+	github.com/prometheus/common v0.55.0 // indirect
```
Comment (nojnhuh) [inline go.mod:175]: I suspect this doesn't affect our ability to drop the `replace` for this module?

Comment (mboersma) [reply go.mod:175]: We could drop the `replace`, but as soon as it moves past v0.47.0 we run into 

```
github.com/prometheus/client_golang/prometheus/testutil/promlint
../../go/pkg/mod/github.com/prometheus/client_golang@v1.18.0/prometheus/testutil/promlint/promlint.go:71:38: undefined: expfmt.FmtText
```

and removing that `replace` just kicks the same error over to component-base. I think it's fixed in component-base v0.31.0 but that's jumping ahead of other k8s vendoring. So it's probably safest to leave the `replace`s intact for now.

Comment (nojnhuh) [reply go.mod:175]: Cool, that's what I was expecting. Thanks for checking!

**Review by nojnhuh — APPROVED** [review]
/lgtm
/approve

**Pull request merged**